### PR TITLE
ecs: merge spellcasting data into class

### PR DESCRIPTION
**Pull request by ecshreve**
## What does this do?
- move spellcasting data to the associated class
- remove the spellcasting json doc
- update `reference` field for spellcasting features in features file

## How was it tested?
- localhost
- ran the db/api locally and issued `curl` requests to various `/class`, `/features` endpoints and validated the responses

## Is there a Github issue this is resolving?
- this addresses one part of https://github.com/bagelbits/5e-database/issues/251

## Did you update the docs in the API? Please link an associated PR if applicable.
- https://github.com/bagelbits/5e-srd-api/pull/130

## Here's a fun image for your troubles
random photo - update me

### CODE REVIEW

File: src/5e-SRD-Features.json [hunk 1]
```diff
@@ -1,8777 +1,8666 @@
...
-    },
-    "subclass": {},
-    "name": "Primal Champion",
-    "level": 20,
-    "desc": [
-      "At 20th level, you embody the power of the wilds. Your Strength and Constitution scores increase by 4. Your maximum for those scores is now 24."
-    ],
-    "url": "/api/features/primal-champion"
-  },
-  {
```
Comment (bagelbits) [inline src/5e-SRD-Features.json]: Why are we removing the features?

Comment (ecshreve) [reply src/5e-SRD-Features.json]: Ahh, this looks like a mistake by me, fixing now

Comment (ecshreve) [reply src/5e-SRD-Features.json]: ahh, i had a json auto-formatter enabled that caused a diff on every line when i saved this file

Comment (bagelbits) [reply src/5e-SRD-Features.json]: I noticed that. :D

**Comment by ogregoire** [comment]
Please don't force-push anything. Just push instead. Now the history is barely understandable (I see @bagelbits responding to a commit 10 minutes before the commit happened). It's okay if you have 50 commits in your PR, we'll squash them all and merge them anyways.

**Comment by ecshreve** [comment]
> Please don't force-push anything. Just push instead. Now the history is barely understandable (I see @bagelbits responding to a commit 10 minutes before the commit happened). It's okay if you have 50 commits in your PR, we'll squash them all and merge them anyways.

Gotcha this makes sense, will keep this in mind going forward. Sorry! New workflow for me 👍

File: src/5e-SRD-Features.json (hunk 1 above)
Comment (bagelbits) [reply src/5e-SRD-Features.json]: Looks like we're still removing these features though?

Comment (bagelbits) [reply src/5e-SRD-Features.json]: Is that on purpose?

Comment (ogregoire) [reply src/5e-SRD-Features.json]: @ecshreve Your PR seems good, except for the part where you remove the features. If we look at the features file, only the warlock is in the proper state. Could you put the other spellcasting classes back in order and only remove the reference field?

Comment (ecshreve) [reply src/5e-SRD-Features.json]: Added 2 new commits to this PR, one reverts the commit that removed spellcasting from the features file, and the second fixes the `reference` urls in each of the spellcasting feature blobs to reference the correct `/classes/<class>/spellcasting` url.

**Review by ogregoire — APPROVED** [review]
This seems perfect. Thank you very much!

**Review by bagelbits — APPROVED** [review]
Awesome! Thank you for doing this! I'll have to make sure these tables get cleaned up in prod.

**Pull request merged**